lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -60,6 +63,56 @@ class HighContrastSupport {
   }
 }
 
+/// Determines the [ui.AppLifecycleState] by listening to 'focus', 'blur', and
+/// 'beforeunload' events on the [DomWindow].
```

Comment:
As you've seen, this platform_dispatcher file is enormous. Do you mind creating a `platform_dispatcher` directory next to it, and add a new `app_lifecycle_state.dart` file defining this class (and the typedef on line 23) that we import and use here?

That helps us keep the file small, and gives us a new place to start moving some of the other classes away from here!